### ISSUE TITLE
encoder in c

### ISSUE THREAD

**Issue by IntendedConsequence** [IntendedConsequence/vadc#6]
- [x] softmax
- [x] layernorm
- [x] batchnorm
- [x] #10
